src/common/structures/api.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { model } from "amf-client-js";
+
+import { resolveApiModel, parseRamlFile, getAllDataTypes } from "../parser";
+
+import { Name } from "./name";
+import { ApiTree } from "./apiTree";
+import { RestApi } from "../../download/exchangeTypes";
+import path from "path";
+import fs from "fs-extra";
+
+/**
+ * An API represented as an AMF model. Includes the extracted data types
+ * defined in the spec. Common transformations of the name are cached for
+ * reference in templates and for use in assembling file paths.
+ */
+export class Api extends ApiTree {
+  dataTypes: model.domain.CustomDomainProperty[];
+  metadata: RestApi;
+  constructor(public model: model.document.Document, filepath: string) {
+    super(
+      new Name((model.encodes as model.domain.WebApi)?.name.value()),
+      filepath,
+      undefined
+    );
+    this.dataTypes = getAllDataTypes(model);
+    Object.freeze(this);
```

Comment:
Should this be applied deeply?

Reply:
This is somewhat broken right now.  Was taken from the original class but doesn't really work.  I have a followup PR to this that will address this.